subsys/bluetooth/mesh/prov_device.c

```diff
@@ -543,17 +553,48 @@ static void prov_data(const uint8_t *data)
 	/* Ignore any further PDUs on this link */
 	bt_mesh_prov_link.expect = PROV_NO_PDU;
 
+	/* Borrow the buffer of prov link to store provision data */
+	prov_data_ptr = (void *)bt_mesh_prov_link.auth;
```

Comment:
`prov_data_ptr` is larger than `auth` array. That means neighbor members of the structure will be rewritten. This is kind of hidden information that after this point part of fields not valid anymore even there were no explicit changes. 
I'm pretty sure `struct bt_mesh_prov_link` will be changed due to upcoming 1.1 features. 
If you need to take a huge piece of memory I suggest to use `conf_inputs` field instead. To the time when `Data` comes 'Confirmation' has been handled and this array is free for reuse.

Reply:
That's what I meant, because auth and the memory behind it will no longer be useful, and I can modify it the way you say.

Reply:
Yep, that's right. My suggestion was to use memory within one array to not change a couple of fields implicitly.